diff --git a/CMakeLists.txt b/CMakeLists.txt
index f9952f8..bcfb540 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,7 +51,7 @@ if(CMAKE_COMPILER_IS_GNUCC)
 	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wno-unused-but-set-variable")
 	if(CMAKE_BUILD_TYPE EQUAL "Release")
 		set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2")
-	else()
+	elseif(CMAKE_BUILD_TYPE STREQUAL "Debug")
 		set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O0")
 	endif()
 	if(WITH_SSE2)
