From d6e3cfc834548385c9047d8a8602931c3d362fdf Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppym@gentoo.org>
Date: Fri, 26 Aug 2016 18:13:25 -0400
Subject: [PATCH] Configure: respect CFLAGS and LDFLAGS

---
 Configure | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/Configure b/Configure
index 3604ba4..aa5aa6a 100755
--- a/Configure
+++ b/Configure
@@ -471,7 +471,7 @@ my $no_sse2=0;
 
 &usage if ($#ARGV < 0);
 
-my $user_cflags="";
+my $user_cflags=$ENV{CFLAGS};
 my @user_defines=();
 $config{openssl_api_defines}=[];
 $config{openssl_algorithm_defines}=[];
@@ -481,7 +481,7 @@ $config{openssl_other_defines}=[];
 my $libs="";
 my $target="";
 $config{options}="";
-$config{build_type} = "release";
+$config{build_type} = "";
 
 my @argvcopy=@ARGV;
 
@@ -906,6 +906,8 @@ my %target = resolve_config($target);
 
 &usage if (!%target || $target{template});
 
+$target{lflags} = $ENV{LDFLAGS};
+
 %target = ( %{$table{DEFAULTS}}, %target );
 
 $target{exe_extension}="";
-- 
2.9.3

