=== modified file 'Makefile.am'
--- Makefile.am	2012-02-10 15:48:48 +0000
+++ Makefile.am	2012-02-21 04:58:43 +0000
@@ -60,9 +60,11 @@
 	sh $(top_srcdir)/geninit.sh `cat $<` > $@ || (rm -f $@; exit 1)
 CLEANFILES += grub_fstest_init.c
 
+grubdata_DATA = grub-mkconfig_lib
+
 if COND_GRUB_MKFONT
 if COND_HAVE_FONT_SOURCE
-grubdata_DATA = unicode.pf2 ascii.pf2 euro.pf2 ascii.h widthspec.h
+grubdata_DATA += unicode.pf2 ascii.pf2 euro.pf2 ascii.h widthspec.h
 endif
 endif
 
@@ -101,8 +103,6 @@
 # Install config.h into platformdir
 platform_HEADERS = config.h
 
-pkgdata_DATA += grub-mkconfig_lib
-
 
 if COND_i386_coreboot
 BOOTTARGET=coreboot

