--- configure.ac
+++ configure.ac
@@ -968,11 +968,16 @@
                         [Define to 1 if you have the LZMA library.])],)
 AC_SUBST([LIBLZMA])
 
-AC_CHECK_LIB([zfs], [libzfs_init],
-             [LIBZFS="-lzfs"
-              AC_DEFINE([HAVE_LIBZFS], [1],
-                        [Define to 1 if you have the ZFS library.])],)
+AC_ARG_ENABLE([zfs],
+			[AS_HELP_STRING([--with-zfs], [Enable ZFS support.])])
+
+if test x"$enable_zfs" != xno; then
+  AC_CHECK_LIB([zfs], [libzfs_init],
+               [LIBZFS="-lzfs"
+                AC_DEFINE([HAVE_LIBZFS], [1],
+                          [Define to 1 if you have the ZFS library.])],)
+fi
 AC_SUBST([LIBZFS])
 
 AC_CHECK_LIB([nvpair], [nvlist_print],
              [LIBNVPAIR="-lnvpair"
