--- a/lib/libcrypto/libmd5/Makefile
+++ b/lib/libcrypto/libmd5/Makefile
@@ -12,7 +12,7 @@
 
 ifeq ($(USE_LIBNSS),true)
 CFLAGS+=-DHAVE_LIBNSS
-CFLAGS+=-I/usr/include/nspr4 -I/usr/include/nss3
+CFLAGS+=$(shell pkg-config --cflags nss)
 endif
 
 include ${srcdir}../../Makefile.library
--- a/lib/libcrypto/libsha1/Makefile
+++ b/lib/libcrypto/libsha1/Makefile
@@ -12,7 +12,7 @@
 
 ifeq ($(USE_LIBNSS),true)
 CFLAGS+=-DHAVE_LIBNSS  
-CFLAGS+=-I/usr/include/nspr4 -I/usr/include/nss3
+CFLAGS+=$(shell pkg-config --cflags nss)
 endif
 
 
--- a/lib/libcrypto/libsha2/Makefile
+++ b/lib/libcrypto/libsha2/Makefile
@@ -15,5 +15,5 @@
 
 ifeq ($(USE_LIBNSS),true)
 CFLAGS+=-DHAVE_LIBNSS
-INCLUDES+=-I/usr/include/nspr4 -I/usr/include/nss3
+INCLUDES+=$(shell pkg-config --cflags nss)
 endif
--- a/lib/libipsecconf/Makefile
+++ b/lib/libipsecconf/Makefile
@@ -33,7 +33,7 @@
 #CFLAGS+=-DFIPS_CHECK
 ifeq ($(USE_LIBNSS),true)
 CFLAGS+=-DHAVE_LIBNSS
-CFLAGS+=-I/usr/include/nspr4 -I/usr/include/nss3
+CFLAGS+=$(shell pkg-config --cflags nss)
 endif
 
 ifeq ($(USE_NM),true)
--- a/lib/libopenswan/Makefile
+++ b/lib/libopenswan/Makefile
@@ -105,7 +105,7 @@
 
 ifeq ($(USE_LIBNSS),true)
 CFLAGS+=-DHAVE_LIBNSS
-CFLAGS+=-I/usr/include/nspr4 -I/usr/include/nss3
+CFLAGS+=$(shell pkg-config --cflags nss)
 endif
 
 ifeq ($(USE_MODP_RFC5114),true)
--- a/programs/Makefile.program
+++ b/programs/Makefile.program
@@ -105,7 +105,7 @@
 
 doinstall:: $(PROGRAM) $(CONFFILES) $(EXTRA8MAN) $(EXTRA5MAN) $(EXTRA5PROC) $(LIBFILES) $(CONFDFILES)
 	# remove any old vendor file installed previously
-	@rm -f $(FINALLIBEXECDIR)/vendor
+	@rm -f $(LIBEXECDIR)/vendor
 	@mkdir -p $(PROGRAMDIR) $(MANDIR8) $(MANDIR5) $(LIBDIR) $(CONFDIR) $(CONFDDIR) $(CONFDDIR)/$(CONFDSUBDIR) $(EXAMPLECONFDIR)
 	@if [ -n "$(PROGRAM)" ]; then $(INSTALL) $(INSTBINFLAGS) $(PROGRAM) $(PROGRAMDIR); fi
 	@$(foreach f, $(addsuffix .8, $(PROGRAM)), \
@@ -179,16 +179,12 @@
 ifeq ($(USE_LIBNSS),false)
 LDLIBS=${LIBS} ${USERLINK} ${LIBS} ${EXTRALIBS} -lgmp
 else
-LDLIBS=${LIBS} ${USERLINK} ${LIBS} ${EXTRALIBS} -lgmp -lnspr4 -lnss3
+LDLIBS=${LIBS} ${USERLINK} ${LIBS} ${EXTRALIBS} -lgmp $(shell pkg-config --libs nss)
 endif
 
 
 %: %.o $(OBJS) ${LIBS}
-ifeq ($(USE_LIBNSS),true)
-	$(CC) $(CFLAGS) -o $@ $@.o ${OBJS} $(LDFLAGS) $(LDLIBS) $(USERLINK) -lnspr4 -lnss3
-else
 	$(CC) $(CFLAGS) -o $@ $@.o ${OBJS} $(LDFLAGS) $(LDLIBS) $(USERLINK)
-endif
 
 # cancel direct version
 %: %.c
--- a/programs/pluto/Makefile.options
+++ b/programs/pluto/Makefile.options
@@ -58,7 +58,7 @@
 
 # Use LIBNSS
 ifeq ($(USE_LIBNSS),true)
-HDRDIRS+= -I/usr/include/nspr4 -I/usr/include/nss3	
+HDRDIRS+= $(shell pkg-config --cflags nss)	
 endif
 
 # BYTE_ORDER = -DBIG_ENDIAN=4321 -DLITTLE_ENDIAN=1234 -DBYTE_ORDER=BIG_ENDIAN
@@ -313,10 +313,7 @@
 ifeq ($(USE_LIBNSS),true)
 HAVE_LIBNSS=1
 DEFINES+=-DHAVE_LIBNSS
-LIBSPLUTO+= -lnspr4
-LIBSPLUTO+= -lnss3
-# For avoiding implicit DSO linking
-LIBSPLUTO+= -lpthread
+LIBSPLUTO+=$(shell pkg-config --libs nss)
 ifeq ($(USE_FIPSCHECK),true)
 FIPS_CHECK=1
 DEFINES+=-DFIPS_CHECK
--- a/programs/rsasigkey/Makefile
+++ b/programs/rsasigkey/Makefile
@@ -21,13 +21,13 @@
 FIPS_CHECK=1
 CFLAGS+=-DFIPS_CHECK
 endif
-INCLUDES+=-I/usr/include/nspr4 -I/usr/include/nss3
+INCLUDES+=$(shell pkg-config --cflags nss)
 endif
 
 PROGRAM=rsasigkey
 LIBS=${OPENSWANLIB} -lgmp
 ifeq ($(USE_LIBNSS),true)
-LIBS+= -lnspr4 -lnss3 -lplc4
+LIBS+=$(shell pkg-config --libs nss)
 ifeq ($(USE_FIPSCHECK),true)
 LIBS+=  -lfipscheck
 endif
--- a/programs/showhostkey/Makefile
+++ b/programs/showhostkey/Makefile
@@ -16,7 +16,7 @@
 include ${OPENSWANSRCDIR}/Makefile.inc
 ifeq ($(USE_LIBNSS),true)
 CFLAGS+=-DHAVE_LIBNSS
-INCLUDES+=-I/usr/include/nspr4 -I/usr/include/nss3
+INCLUDES+=$(shell pkg-config --cflags nss)
 endif
 
 PROGRAM=showhostkey
