--- pithos/pithosconfig.py	2010-11-16 16:33:23.000000000 -0600
+++ pithos/pithosconfig.new.py	2010-11-16 16:34:33.000000000 -0600
@@ -16,7 +16,7 @@
 
 # where your project will head for your data (for instance, images and ui files)
 # by default, this is ../data, relative your trunk layout
-__pithos_data_directory__ = '../data/'
+__pithos_data_directory__ = '/usr/share/pithos/'
 __license__ = 'GPL-3'
 
 VERSION = '0.3.5'